### PR TITLE
[docs] Update TransparentAttr.rst for '@inlinable'

### PR DESCRIPTION
**Pull request by jrose-apple**

### CODE REVIEW

File: docs/TransparentAttr.rst [hunk 1]
```diff
@@ -14,13 +14,13 @@ This has several consequences:
...
 
-- Because of this, a ``@_transparent`` function is inherently "fragile", in
+- Because of this, a ``@_transparent`` function is implicitly inlinable, in
   that changing its implementation most likely will not affect callers in
   existing compiled binaries.
 
 - Because of this, a ``@_transparent`` function MUST only reference public
   symbols, and MUST not be optimized based on knowledge of the module it's in.
-  [This is not currently implemented or enforced.]
+  [The former is caught by checks in Sema.]
```
Comment (slavapestov) [inline docs/TransparentAttr.rst]: Both of these are only true for public transparent functions. Internal and private transparent can reference private symbols.

**Review by slavapestov — APPROVED** [review]

File: docs/TransparentAttr.rst (hunk 1 above)
Comment (jrose-apple) [reply docs/TransparentAttr.rst]: Ah, right, thanks.

**Review by airspeedswift — APPROVED** [review]
Worth mentioning two other things that came up recently?
- it's pointless (though not banned) to mark something `@inlinable` if it's also `@_transparent`
- unlike `@inlineable`, `@_transparent` doesn't imply `@usableFromInline`

**Review by ravikandhadai — APPROVED** [review]

**Comment by jrose-apple** [comment]
Updated for feedback.

@swift-ci Please smoke test

**Review by shajrawi — APPROVED** [review]

**Pull request merged**